hexa/workspaces/schema/mutations.py

```diff
@@ -122,57 +122,61 @@ def resolve_invite_workspace_member(_, info, **kwargs):
             user = User.objects.get(email=input["userEmail"])
-            workspace_membership = WorkspaceMembership.objects.create_if_has_perm(
-                principal=request.user,
-                workspace=workspace,
-                user=user,
-                role=input["role"],
-            )
+            # If the user already exists, check if his not already a member of the workspace
+            if WorkspaceMembership.objects.filter(
+                user__email=input["userEmail"], workspace=workspace
+            ).exists():
+                raise AlreadyExists
+
             if not user.has_feature_flag("workspaces"):
                 FeatureFlag.objects.create(
                     feature=Feature.objects.get(code="workspaces"), user=user
                 )
 
+            invitation = WorkspaceInvitation.objects.create_if_has_perm(
+                principal=request.user,
+                workspace=workspace,
+                email=input["userEmail"],
+                role=input["role"],
+            )
+
             with override(user.language):
                 send_mail(
                     title=gettext_lazy(
-                        "You've been added to the workspace {workspace_name}"
+                        "You've been invited to join the workspace {workspace_name}"
                     ).format(workspace_name=workspace.name),
                     template_name="workspaces/mails/invite_member",
                     template_variables={
                         "workspace": workspace.name,
                         "owner": request.user.display_name,
-                        "workspace_url": request.build_absolute_uri(
-                            f"//{settings.NEW_FRONTEND_DOMAIN}/workspaces/{workspace.slug}"
+                        "account_url": request.build_absolute_uri(
+                            f"//{settings.NEW_FRONTEND_DOMAIN}/user/account"
                         ),
                     },
                     recipient_list=[user.email],
                 )
             return {
                 "success": True,
                 "errors": [],
-                "workspace_membership": workspace_membership,
             }
         except User.DoesNotExist:
-            try:
-                invitation = WorkspaceInvitation.objects.create_if_has_perm(
-                    principal=request.user,
-                    workspace=workspace,
-                    email=input["userEmail"],
-                    role=input["role"],
-                )
-                send_workspace_invitation_email(invitation)
-                return {
-                    "success": True,
-                    "errors": [],
-                }
-            except IntegrityError:
-                return {
-                    "success": False,
-                    "errors": ["ALREADY_EXISTS"],
-                }
+            invitation = WorkspaceInvitation.objects.create_if_has_perm(
```

Comment:
This is exactly what you had above. It would be nice to refactor this mutation to simplify it.

`send_workspace_invitation_email` has to be used for user and non-user. You can do the check for the user in the body of the function (to set the correct language if possible)